### DISCUSSION TITLE
Easy Configuration with Tailwind CSS

### DISCUSSION THREAD

**Discussion by NimaSoltanM** [honojs/hono#2950] [Ideas]
Dear Hono developers and community,
I am writing to request a feature that would make it easier to integrate Tailwind CSS with Hono.js. I've been using HTMX and Hono for a while and really start to notice this thing. Currently, adding Tailwind CSS to a Hono.js project requires manual configuration, which can be time-consuming and error-prone. This can be a significant barrier for developers who are new to Hono.js or are not familiar with the configuration process.
To make it easier for users to add Tailwind CSS, I suggest implementing a simple command that would automatically configure Tailwind CSS for a Hono.js project. This would save developers a significant amount of time and effort, allowing them to focus on building their applications rather than wrestling with configuration files.
Here is an example of how this command could work:
Run hono tailwind init to initialize the Tailwind CSS configuration.
This single command would create the necessary files, import the Tailwind CSS styles, and configure the CSS preprocessors, making it easy for developers to get started with Tailwind CSS in their Hono.js projects.
I believe that this feature would be a significant improvement for many users and would make Hono.js more competitive with other frameworks. Thank you for your time and consideration. I look forward to hearing your thoughts on this proposal.
Best regards,

**Comment by yusukebe** [comment]
Regarding Tailwind CSS, I think it would be great if we had a Vite plugin to enable and set it up easily.

@NimaSoltanM Are you using Vite for Tailwind CSS?

**Reply by NimaSoltanM** [reply]
Yes

**Reply by yusukebe** [reply]
Cool. It would be best if we could do this:

```ts
import { defineConfig } from 'vite'
import tailwind from '@hono/vite-tailwind-css'

export default defineConfig({
  plugins: [tailwind()]
})
```

**Comment by jimbits** [comment]
How would this get written? I am in if anyone wants to lead

**Comment by grunklejp** [comment]
This is probably unnecessary now, the tailwind CLI is very easy to use.

https://tailwindcss.com/docs/installation/tailwind-cli

**Comment by yusukebe** [comment]
Yes. Currently, TailwindCSS has the Vite plugin. You can use it:

https://tailwindcss.com/docs/installation/using-vite
